tests.py

```diff
@@ -368,3 +386,116 @@ def test_version(self, testbot):
         assert '1.0.0' in response
 
 
+@pytest.fixture(scope="module")
+def github_api():
+    return GithubAPI()
+
+
+def test_github_fetch_repos(github_api, mocker):
+    github_mock = mocker.patch.object(github_api, '_github', autospec=True)
+    repos = ['conda-devenv', 'deps', 'alfasim-sdk']
+    github_mock.get_organization.return_value.get_repos.return_value = repos
+
+    assert github_api.fetch_repos("esss") == repos
+    github_mock.get_organization.assert_called_with('esss')
+
+
+def test_github_fetch_branches(github_api, mocker):
+    github_mock = mocker.patch.object(github_api, '_github', autospec=True)
+    github_mock.get_organization.return_value.get_repo.return_value.get_branch.return_value = \
+        'single-branch'
+
+    github_mock.get_organization.return_value.get_repo.return_value.get_branches.return_value = \
```

Comment:
Are you setting the return values?

Reply:
Yes, it's hard to test Github and Stash APIs without abusing of mocks. I've tried testing using the API calls, but I ended up getting `Github API rate limit exceeded`.

Reply:
It is not an abuse. Making network calls when not strictly necessary is bad.